include/seqan3/search/algorithm/configuration/max_error_rate.hpp

```diff
@@ -0,0 +1,142 @@
+
+/*!\file
+ * \brief Provides the configuration for maximum number of errors in percent to the query length across all error types.
+ * \author Christopher Pockrandt <christopher.pockrandt AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <string>
+
+#include <seqan3/core/algorithm/all.hpp>
+#include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+#include <seqan3/search/algorithm/configuration/utility.hpp>
+
+namespace seqan3::detail
+{
+/*!\brief A configuration element for the maximum number of errors in percent to the query length across all error types
+          (mismatches, insertions, deletions). This is an upper bound of errors independent from error numbers or rates
+          of specific error types.
+ * \ingroup search_configuration
+ */
+struct search_config_max_error_rate
+{
+    //!\brief The actual value.
+    std::tuple<double, double, double, double> value;
+};
+
+/*!\brief The max_error_rate adaptor enabling pipe notation.
+ * \ingroup search_configuration
+ */
+struct search_config_max_error_rate_adaptor :
+    public configuration_fn_base<search_config_max_error_rate_adaptor>
+{
+
+    /*!\brief Adds to the configuration a max_error configuration element.
+     * \relates seqan3::search_config_max_error
+     * \param[in] cfg  The configuration to be extended.
+     * \param[in] total_error The number of maximum errors over all error types.
+     * \param[in] substitution_error The number of maximum substitution errors.
+     * \param[in] insertion_error The number of maximum insertion errors.
+     * \param[in] deletion_error The number of maximum deletion errors.
+     * \returns A new configuration containing the max_error configuration element.
+     */
+    template <typename configuration_t>
+    //!\cond
+        requires is_algorithm_configuration_v<remove_cvref_t<configuration_t>>
+    //!\endcond
+    // TODO: replace int by templates and check whether it can be casted to uint8_t
+    // TODO: template argument packing. allow any order and subset of strong types
+    constexpr auto invoke(configuration_t && cfg, search_cfg::total<double> const total_error,
+                                                  search_cfg::substitution<double> const substitution_error,
+                                                  search_cfg::insertion<double> const insertion_error,
+                                                  search_cfg::deletion<double> const deletion_error) const
+    {
+        static_assert(is_valid_search_configuration_v<search_cfg::id::max_error, remove_cvref_t<configuration_t>>,
+                      SEQAN3_INVALID_CONFIG(search_cfg::id::max_error));
+
+        double total {total_error},
+               substitution {substitution_error},
+               insertion {insertion_error},
+               deletion {deletion_error};
+
+        if (0 > total        || total > 1 ||
```

Comment:
strong types have a comparison skill.

Reply:
strong_type_skill::logic

Reply:
You didn't address this?

Reply:
I can only find arithmetic skills and bitwise operations there

Reply:
I'd leave this for now. This part will be refactored soon anyway, there's a TODO above the functions signature.

Reply:
Oh, you are right. we are missing comparison operators there.
Later